src/http/http.client.ts

```diff
@@ -52,15 +51,23 @@ class Client {
 
     const timeRequest = httpHistogram.startTimer(labels);
 
-    // add catch for instrumentation
-    return this.transport.request<T>(options).then((response: AxiosResponse<T>) => {
-      labels.status = response.status;
-      timeRequest();
-      return response;
-    }).catch((error) => {
+    let response: AxiosResponse<T>;
+
+    try {
+      response = await this.transport.request<T>(options);
+    } catch (error) {
       timeRequest();
       throw new TransportError(error.message);
-    });
+    }
+
+    labels.status = response.status;
+    timeRequest();
+
+    if (response.status >= 400) {
+      throw new (getErrorFromStatusCode(response.status) as any);
```

Comment:
Why as any?

Reply:
because the `new` wasn't allowed to instantiate the return of a function :(

Reply:
is there a way to create an instance of the `BaseError` without using `any`?